src/pineko/parser.py

```diff
@@ -35,7 +71,54 @@ def _load_yaml(yaml_file):
     return ret
 
 
-def get_yaml_information(yaml_file, grids_folder):
+def _apfelcomb_compatibility_flags(gridpaths, metadata):
+    """
+    Prepare the apfelcomb-pineappl compatibility fixes
+
+    Returns
+    -------
+        apfelcomb_norm: np.array
+            Per-point normalization factor to be applied to the grid
+            to be compatible with the data
+        apfelcomb_repetition_flag: bool
+            Whether the fktable is a single point which gets repeated up to a certain size
+            (for instance to normalize a distribution)
+        shift: list(int)
+            Shift in the data index for each grid that forms the fktable
+    """
+    operands = metadata["operands"]
+    apfelcomb_norm = None
+    if norms := metadata.get("apfelcomb_norm"):
```

Comment:
Please, avoid walrus...

Reply:
And test for `bool`, not for truthiness...
(i.e. make the explicit check: if you expect float, test for a float, if you simply want to test if the field is present, test for `is not None`)

Reply:
> Please, avoid walrus...

Why?

Reply:
> Why?

It deteriorates readability. Having side effects on assignment makes less intuitive which is the actual expression (plus is easily avoidable 3.8 feature, we gain nothing to little, and we pointlessly reduce compatibility with older Python versions, but this is a minor detail, we explicitly do not support <3.8 in any case).

Reply:
However, it's a detail in any case, so do as you wish.

Reply:
I think I agree with @AleCandido : side-effects are a dangerous C heritage (also agree that it's a detail)

Reply:
Well, 2 against 1. I'll remove it. But [I'm crying](https://www.youtube.com/watch?v=TKuvJLTeJYY)